ci/run_python_tests.sh

```diff
@@ -0,0 +1,183 @@
+  fn_die "Error: TEST_CMD variable is not set. Exiting ..."
+fi
+
+if [ -z "${TEST_ARGS:-}" ]; then
+  fn_die "Error: TEST_ARGS variable is not set. Exiting ..."
+fi
+
+if [ -z "${API_ZEN_REPO_URL:-}" ]; then
+  fn_die "Error: API_ZEN_REPO_URL variable is not set. Exiting ..."
+fi
+
+if [ -z "${ZEN_REPO_MAINTAINER_KEYS:-}" ]; then
+  fn_die "Error: ZEN_REPO_MAINTAINER_KEYS variable is not set. Exiting ..."
+fi
+
+CURRENT_DIR="${PWD}"
+
+# Step 1
+echo "" && echo "=== GGet latest ZEN repo PROD build id and commit hash ===" && echo ""
+
+zen_tag="$(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/HorizenOfficial/zen/git/refs/tags | jq -r '[.[] | select(.ref | test("refs/tags/v[0-9]\\.[0-9]\\.[0-9]$"))][-1].ref' | sed -e 's|refs/tags/||')"
+check_runs="$(curl -sL -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" "${API_ZEN_REPO_URL}/commits/${zen_tag}/check-runs")"
+travis_build_id="$(basename "$(jq -rc '.check_runs[0].details_url' <<< "${check_runs}")")"
+commit_sha="$(jq -rc '.check_runs[0].head_sha' <<< "${check_runs}")"
+
+travis_urls="
+#amd64
+https://f001.backblazeb2.com/file/ci-horizen/amd64-linux-ubuntu_focal-${travis_build_id}-${commit_sha}.tar.gz.sha256
+https://f001.backblazeb2.com/file/ci-horizen/amd64-linux-ubuntu_focal-${travis_build_id}-${commit_sha}.tar.gz
+"
+
+
+# Step 2
+echo "" && echo "=== Create folder structure ===" && echo ""
+
+base_dir="${CURRENT_DIR}/zen_release"
+is_cached=false
+
+if [ -d "${base_dir}/travis_files" ]; then
+  echo "${base_dir} folder already exists, using cache!"
+  cd "${base_dir}"/travis_files
+  is_cached=true
+else
+  mkdir -p "${base_dir}"/{travis_files,src}
+
+  # Step 3
+  echo "" && echo "=== Downloading ZEN artifacts from remote bucket ===" && echo ""
+
+  cd "${base_dir}"/travis_files
+  echo "$travis_urls" > ./travis_urls.txt
+  sudo apt-get update
+  sudo apt-get -y --no-install-recommends install aria2
+  aria2c -x16 -s16 -i ./travis_urls.txt --allow-overwrite=true --always-resume=true --auto-file-renaming=false
+fi
+
+
+# Step 4
+echo "" && echo "=== Checksum verification ===" && echo ""
+
+if shasum -a256 -c ./*.sha256; then
+  echo "Checksum verification passed."
+else
+  fn_die "Error: checksum verification failed. Exiting ..."
+fi
+
+# Step 5
+release_folder="zen-${zen_tag}-amd64"
+if [ "$is_cached" = false ]; then
+  echo "" && echo "=== Extract artifacts from tar ===" && echo ""
+  tar_file="$(find "$(realpath ${base_dir}/travis_files/)" -type f -name "*.tar.gz")"
+
+  mkdir -p "${base_dir}/src/${release_folder}"
+  tar -xzf "${tar_file}" -C "${base_dir}/src/${release_folder}"
+fi
+
+# Step 6
+echo "" && echo "=== Verify git tag signed by allowlisted maintainer ===" && echo ""
+
+cd "${base_dir}/src/${release_folder}"
+GNUPGHOME="$(mktemp -d 2>/dev/null || mktemp -d -t "GNUPGHOME")"
+export GNUPGHOME
+import_gpg_keys "${ZEN_REPO_MAINTAINER_KEYS}"
+check_signed_tag "${zen_tag}" && IS_RELEASE="true" || IS_RELEASE="false"
```

Comment:
If you sort out IS_RELEASE vs IS_A_RELEASE under the function you will NOT need any logic after you call the function starting from **&&**

Also I see you are setting IS_RELEASE var but where do you actually use it in the end after export?

Reply:
see comment above.